sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  // WARNING: The type "Models.AppendBlobAppendBlockFromUrlResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.AppendBlobAppendBlockFromUrlResponse" needs to be exported by the package (e.g. added to index.ts)
+  appendBlockFromURL(sourceURL: string, sourceOffset: number, count: number, options?: AppendBlobAppendBlockFromURLOptions): Promise<Models.AppendBlobAppendBlockFromUrlResponse>;
+  // WARNING: The type "Models.AppendBlobCreateResponse" needs to be exported by the package (e.g. added to index.ts)
+  create(options?: AppendBlobCreateOptions): Promise<Models.AppendBlobCreateResponse>;
+  withSnapshot(snapshot: string): AppendBlobClient;
+}
+
+// @public
+interface AppendBlobCreateOptions {
+  abortSignal?: AbortSignalLike;
+  accessConditions?: BlobAccessConditions;
+  blobHTTPHeaders?: Models.BlobHTTPHeaders;
+  customerProvidedKey?: Models.CpkInfo;
+  metadata?: Metadata;
+}
+
+// @public (undocumented)
+class BaseRequestPolicy implements RequestPolicy {
+  protected constructor(_nextPolicy: RequestPolicy, _options: RequestPolicyOptions);
+  // (undocumented)
+  readonly _nextPolicy: RequestPolicy;
+  // (undocumented)
+  readonly _options: RequestPolicyOptions;
+  log(logLevel: HttpPipelineLogLevel, message: string): void;
+  // (undocumented)
+  abstract sendRequest(webResource: WebResource): Promise<HttpOperationResponse>;
+  shouldLog(logLevel: HttpPipelineLogLevel): boolean;
+}
+
+// @public
+class BatchDeleteRequest extends BatchRequest {
+  constructor();
+  addSubRequest(url: string, credential: SharedKeyCredential | AnonymousCredential | TokenCredential, options?: BlobDeleteOptions): Promise<void>;
+}
+
+// @public
+class BatchRequest {
+  constructor();
+  // (undocumented)
+  protected addSubRequestInternal(subRequest: BatchSubRequest, assembleSubRequestFunc: () => Promise<void>): Promise<void>;
+  // (undocumented)
+  protected readonly batch: string;
+  // WARNING: The type "InnerBatchRequest" needs to be exported by the package (e.g. added to index.ts)
+  // (undocumented)
+  protected batchRequest: InnerBatchRequest;
+  getHttpRequestBody(): string;
+  getMultiPartContentType(): string;
+  getSubRequests(): Map<number, BatchSubRequest>;
+}
+
+// @public
+class BatchSetTierRequest extends BatchRequest {
+  constructor();
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.AccessTier" needs to be exported by the package (e.g. added to index.ts)
+  addSubRequest(url: string, credential: SharedKeyCredential | AnonymousCredential | TokenCredential, tier: Models.AccessTier, options?: BlobSetTierOptions): Promise<void>;
+}
+
+// @public (undocumented)
+interface BatchSubRequest {
+  credential: SharedKeyCredential | AnonymousCredential | TokenCredential;
```

Comment:
maybe a type alias for credential would be convenient, since the union is used in at least two places?

Reply:
The union is nice because it shows the types a user might instantiate to pass in directly in the signature. I like this aspect :)

Reply:
Arch board likes unions being exposed.

Reply:
(If the expanded union is self-explanatory)